core/templates/dev/head/components/gadget_panel.html

```diff
@@ -1,6 +1,6 @@
 <script type="text/ng-template" id="components/gadgetPanel">
-  <div ng-repeat="gadgetSpec in panelContents()">
-    <oppia-gadget class="protractor-test-<[gadgetSpec.gadget_type]>-gadget" gadget-type="gadgetSpec.gadget_type" gadget-customization-args="gadgetSpec.customization_args" show-in-states="gadgetSpec.visible_in_states">
+  <div ng-repeat="gadgetInstance in panelContents()">
+    <oppia-gadget class="protractor-test-<[gadgetInstance.gadget_type]>-gadget" gadget-type="gadgetInstance.gadget_type" gadget-name="gadgetInstance.gadget_name" gadget-customization-args="gadgetInstance.customization_args" show-in-states="gadgetInstance.visible_in_states">
```

Comment:
Should the protractor-test-* class refer to the name instead of the type? (Since names will be unique, but types won't be?)